src/zenml/orchestrators/step_runner.py

```diff
@@ -415,3 +509,38 @@ def _store_output_artifacts(
             )
             output_artifacts[output_name] = output_artifact
         return output_artifacts, output_artifact_metadata
+
+    def _run_hook(
+        self,
+        hook_source: str,
+        step_exception: Optional[Exception],
+        output_artifact_uris: Dict[str, str],
+        output_materializers: Dict[str, Type[BaseMaterializer]],
+    ) -> None:
+        """Runs a step hook.
+
+        Args:
+            hook_source: The source of the hook function.
+            step_exception: The exception of the original step.
+            output_artifact_uris: The URIs of the output artifacts of the step.
+            output_materializers: The output materializers of the step.
+
+        Raises:
+            ValueError: When failed to load hook source.
+        """
+        try:
+            hook = source_utils.load_source_path(hook_source)
+            hook_spec = inspect.getfullargspec(inspect.unwrap(hook))
+            function_params = self._parse_hook_inputs(
+                args=hook_spec.args,
+                annotations=hook_spec.annotations,
+                step_exception=step_exception,
+                output_artifact_uris=output_artifact_uris,
+                output_materializers=output_materializers,
+            )
+            logger.debug(f"Running hook {hook} with params: {function_params}")
+            hook(**function_params)
+        except (ImportError, TypeError, ValueError) as e:
+            raise ValueError(
```

Comment:
This `try...except` should only be around the actual `load_source_path` call if it is to catch the loading exceptions.
IMO we want a general
```python
try:
  load_and_run_hook()
except:
  logger.error("Failed to run hook")
```

here instead though. Failure to run a hook should never actually lead to a failure of the pipeline